### PR TITLE
Send all on withdraw if necessary

### LINKED ISSUE #355: Handle withdraw of all funds
Send all L-BTC funds must take into account the L-BTC needed for fees.

Getting error `need unspents for the asset: 6f0279e9ed041c3d710a9f57d0c02928416460c4b722ae3457a11eec381c526d`

### PR DESCRIPTION
**Pull request by Janaka-Steph**
Update tdex-sdk to 1.7.8 / LDK 0.3.16

Use `sendTx` to withdraw, which will subtract fee if necessary. 
It closes #355 

Please review @tiero @louisinger

### CODE REVIEW

File: src/pages/Withdrawal/index.tsx [hunk 1]
```diff
@@ -179,29 +168,35 @@ const Withdrawal: React.FC<WithdrawalProps> = ({
       } catch (_) {
         throw IncorrectPINError;
       }
+      // Check if substract fee necessary
+      const LbtcBalance = balances.find((b) => b.coinGeckoID === 'bitcoin')?.amount || 0;
+      const fee = estimateFeeAmount(utxos, [getRecipient()], 0.1, () => null);
+      const needLBTC = toSatoshi(LbtcBalance.toString(), 8, lbtcUnit).plus(fee).toNumber();
+      const substractFee = needLBTC > LbtcBalance;
```
Comment (louisinger) [inline src/pages/Withdrawal/index.tsx:175]: This check is not necessary. My bad if the argument name in sendTx is not enough clear. You must read it as "subtract fee **if and only if** the user inputs an amount greater than available balance". It means that this check is done inside the `sendTx` function. In other words, the `substractFee` var [should be a constant set to true.](https://github.com/vulpemventures/marina/pull/244/commits/9680c4f0e4ebefc534069163f78afb636f26008c#diff-a9c34e278ae2836b9ea141dd8456cc9157aed7fd7a0e0d0b18fcc7fb1ea2c16cR284)

Comment (Janaka-Steph) [reply src/pages/Withdrawal/index.tsx:175]: Ha ok ok

Comment (Janaka-Steph) [reply src/pages/Withdrawal/index.tsx]: done

File: src/components/WithdrawRow/index.tsx [hunk 2]
```diff
@@ -124,19 +124,13 @@ const WithdrawRow: React.FC<WithdrawRowInterface> = ({ amount, balance, price, s
...
-              '0.00'
-            ) : (
-              <span>
-                {`${residualBalance && residualBalance} ${
-                  balance.ticker === 'L-BTC' ? lbtcUnit : balance.ticker.toUpperCase()
-                }`}
-              </span>
-            )}
-          </>
+        <span className="balance" onClick={() => setAmount(residualBalance)}>
```
Comment (tiero) [inline src/components/WithdrawRow/index.tsx:127]: is this taking into account LBTC units?

File: src/pages/TradeSummary/index.tsx [hunk 3]
```diff
@@ -138,7 +138,7 @@ const TradeSummary: React.FC<TradeSummaryProps> = ({ history, location }) => {
                         <div className="transaction-info-date">
                           {transaction && <span>{transaction.blockTime?.format('DD MMM YYYY HH:mm:ss')}</span>}
                           {transaction ? (
-                            <span>{fromSatoshiFixed(transaction.fee, 8, 8)} Fee</span>
+                            <span>{fromSatoshiFixed(transaction.fee.toString(), 8, 8)} Fee</span>
```
Comment (tiero) [inline src/pages/TradeSummary/index.tsx:141]: why we need to change this now?

File: src/utils/coinSelection.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import type { CoinSelectorErrorFn } from 'tdex-sdk';
+
+export const throwErrorHandler: CoinSelectorErrorFn = (asset: string, need: number, has: number) => {
+  throw new Error(`not enough funds to fill ${need}sats of ${asset} (amount selected: ${has})`);
```
Comment (tiero) [inline src/utils/coinSelection.ts:4]:
```suggestion
  throw new Error(`not enough funds to fill ${need}sats of ${asset} (available: ${has})`);
```

File: src/pages/TradeSummary/index.tsx (hunk 3 above)
Comment (Janaka-Steph) [reply src/pages/TradeSummary/index.tsx:141]: After fixing Reselect typing bug this TS typing error appeared. fromSatoshiFixed expect a string and transaction.fee is a number

File: src/components/WithdrawRow/index.tsx (hunk 2 above)
Comment (Janaka-Steph) [reply src/components/WithdrawRow/index.tsx:127]: Yes

File: src/pages/TradeSummary/index.tsx (hunk 3 above)
Comment (Janaka-Steph) [reply src/pages/TradeSummary/index.tsx:141]: I double checked and it doesn't break things

**Review by tiero — APPROVED** [review]
utACK

**Pull request merged**